Tests/ProjectDescriptionTests/Up/UpMintTests.swift

```diff
@@ -0,0 +1,15 @@
+import Foundation
+import XCTest
+
+@testable import ProjectDescription
+@testable import TuistSupportTesting
+
+final class UpMintTests: XCTestCase {
+    func test_codable_when_all() {
+        // Given
+        let subject = UpMint(linkPackagesGlobally: true)
+
+        // Then
+        XCTAssertCodable(subject)
```

Comment:
This test is failing due to:

```swift
// Up.swift

public static func == (_: Up, _: Up) -> Bool {
        fatalError("Subclasses should override this method")
}
```

Looks like there might be a problem with overriding an `Equatable.==` function by the child classes. Here is a thread on Swift forum about the issue: https://forums.swift.org/t/implement-equatable-protocol-in-a-class-hierarchy/13844. 

Looking  👀

Reply:
I gave a go of this 👇 

> Instead, you can have == call out to an instance method, perhaps equals(_:), and then override that. This is what Foundation's NSObject type does.

Reply:
That looks perfect!